### DISCUSSION TITLE
[Feature] duplicate detection blacklist

### DISCUSSION THREAD

**Discussion by spawnpeekboi** [immich-app/immich#12633] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

duplicated detect blacklist on criteria, like in a certain album, or being selected to kept before.
sometimes I just want every photo in a certain album. like moments in wedding photo. I do not want them to be stacked. However, each time I run duplicate detect, they just being select out again and as me to keep or not. 
So if there is a option that i could blacklist a certain selection of image being pick to be deduplicated that will be very nice
thankyou

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by maicss** [comment]
https://github.com/immich-app/immich/discussions/17566

**Comment by orrd** [comment]
This same feature has been requested several times, but I think this one is the oldest request for it?  (But someone should probably fix the spelling/wording of the title of it.)  

There does need to be some mechanism to remember when the user wants duplicates to be ignored so they don't keep on re-appearing in the duplicates utility.  The way I implemented this on a project I worked on was to have a "nonduplicates" table that records sets of photos that have already been reviewed.  So for Immich, each time the user reviews a set of duplicates and chooses to keep more than one, it would add a row to the nonduplicates table.  Then that set of nonduplicates would be ignored on future duplicate searches.  

If a new photo is added that matches past photos that were nonduplicates, it's ok for it to show the group of photos again with the new photo added.  Using the "all" button on the duplicate detection item on the Job Status page should also delete all rows of the nonduplicates table, since that is a way to make it reconsider all duplicates again.  

I don't have experience with this code base, so I'm not sure if it makes sense for me to try to tackle this or not.  I also don't know if anyone is already working on a solution for this.

**Comment by orrd** [comment]
This appears to have been fixed in 1.133 or 1.133.1.  I don't see any mention of anything like this in the release notes, but it appears to work now!

**Reply by fabb** [reply]
The issue still persists for me in 2.4.1

**Reply by orrd** [reply]
> The issue still persists for me in 2.4.1

I just tried it and it's working for me.  One possible bit of confusion, if you use the "next" button when going through the photos, they'll continue to show up again.  You have to use the buttons on the top right where it says "keep", "trash", etc to make it not show the same duplicates again.

**Reply by fabb** [reply]
I did that. 

When reproducing, did you manually run the `Duplicate Detection` job with the `All` button?

![IMG_2582](https://github.com/user-attachments/assets/daf8c3e8-f15f-4672-af6d-0d126b59eb3c)

**Reply by orrd** [reply]
No, I don't want to do that because I think that would make it reset everything and I would have to do all the duplicates again.  But if you go to the Duplicates utility, it shouldn't show you ones you've already done.  Unless you make it run duplication detection on all the photos again.

**Reply by fabb** [reply]
It does not reset everything, it remembers all stacked photos, and deleted photos are deleted anyways. Just `Keep all` photos are showing up as duplicates again.

**Reply by orrd** [reply]
Yes, that's true. But I have tons of photos where I want to keep them as is.  So I think it's all working as it should be, yes?

**Reply by fabb** [reply]
Hm i‘d say if i chose to keep certain duplicates, the decision should be remembered even when running the Duplicate Detection job again. The question is how often it is really necessary to run that job again.

**Reply by orrd** [reply]
I believe that behavior was intentional, there are times when someone may want it to show all the existing duplicates again, and that "All" button is a way to make it do that.  But I don't know, I'm just guessing.
